### PR TITLE
[RFR] 07 - announcement details 

### PR DESCRIPTION
**Pull request by mRoca**
Ref #17

### CODE REVIEW

File: src/SensioLabs/JobBoardBundle/Resources/config/services.xml [hunk 1]
```diff
@@ -1,27 +1,41 @@
 <?xml version="1.0" ?>
 
 <container xmlns="http://symfony.com/schema/dic/services"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
```
Comment (saro0h) [inline src/SensioLabs/JobBoardBundle/Resources/config/services.xml]: You should undo those changes.

File: src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig [hunk 2]
```diff
@@ -8,12 +8,12 @@
     {% block job_description %}
         <div class="clear"></div>
         <div id="job-description" class="padding description">
-           {{ job.description | purify }}
+            {{ job.description | purify }}
```
Comment (saro0h) [inline src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig]: Extra white space.

File: src/SensioLabs/JobBoardBundle/Resources/config/services.xml (hunk 1 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Resources/config/services.xml]: Done

File: src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig (hunk 2 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig]: Fixed

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php [hunk 3]
```diff
@@ -55,6 +56,50 @@ public function testPreviewAction()
         $this->assertEquals(1, $crawler->filter('body.preview')->count());
     }
 
+    public function testUpdateAction()
+    {
+        $client = static::createClient();
```
Comment (florianbelhomme) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: All the tests are using a client, you can create the client in the setup()

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php [hunk 4]
```diff
@@ -55,6 +56,50 @@ public function testPreviewAction()
...
+        $em = $client->getContainer()->get('doctrine')->getManager();
+        $job = $em->getRepository('SensioLabsJobBoardBundle:Job')->findOneBy(array('user' => $this->user));
+
+        if (!$job instanceof Job) {
+            throw new EntityNotFoundException('Entity not found');
```
Comment (florianbelhomme) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: It should be an assert, the expected return IS a job ( => assertInstanceOf )

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php [hunk 5]
```diff
@@ -55,6 +56,50 @@ public function testPreviewAction()
...
+        $this->assertEquals(1, $crawler->filter('body.add')->count());
+
+        $form = $crawler->selectButton('Save')->form();
+
+        $this->assertEquals($job->getTitle(), $form->get('job[title]')->getValue());
+
+        $client->submit($form, $this->jobData());
+        $crawler = $client->followRedirect();
+
+        $this->assertEquals(1, $crawler->filter('body.preview')->count());
```
Comment (florianbelhomme) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: => assertCount https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertCount The returned messages are better if you use the specific assert

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php [hunk 6]
```diff
@@ -55,6 +56,50 @@ public function testPreviewAction()
...
+    public function testShowAction()
+    {
+        $client = static::createClient();
+        $crawler = $client->request('GET', '/');
+
+        $this->assertGreaterThan(0, $crawler->filter('#job-container > .box a.title')->count());
+        $link = $crawler->filter('#job-container > .box a.title')->first()->link();
+
+        $client->request('GET', $link->getUri());
+        $this->assertTrue($client->getResponse()->isSuccessful());
```
Comment (florianbelhomme) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: $this->assertSame(Response::HTTP_OK, $client->getResponse()->getStatusCode());  Will make a message like 'excepted 200, got 404', better that 'Excepted true'

Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: Done

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php (hunk 5 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: Done

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php (hunk 4 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: Done

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php (hunk 3 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: Done

File: src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig (hunk 2 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig]: It was missing

**Comment by alOneh** [comment]
Can you rebase this ?

**Comment by mRoca** [comment]
Rebased

**Pull request merged**